File: 2023/advent_of_code/test/q3/q3_test.exs
defmodule AdventOfCode.Q3Test do
  use ExUnit.Case
  doctest AdventOfCode.Q3

  test "Q3 Part 1 Example works" do
    file_stream = File.stream!("./test/q3/q3_example.txt")
    result = AdventOfCode.Q3.part1(file_stream)
    assert result == 4361
  end

  test "Q3 Part 1 works" do
    file_stream = File.stream!("./test/q3/q3.txt")
    result = AdventOfCode.Q3.part1(file_stream)
    assert result == 535_078
  end

  test "Q3 Part 2 Example works" do
    file_stream = File.stream!("./test/q3/q3_example.txt")
    result = AdventOfCode.Q3.part2(file_stream)
    assert result == 467_835
  end

  test "Q3 Part 2 works" do
    file_stream = File.stream!("./test/q3/q3.txt")
    result = AdventOfCode.Q3.part2(file_stream)
    assert result == 75_312_571
  end
end

File: 2022/AdventOfCode/src/test/kotlin/q3/Q3Test.kt
package q3

import java.nio.file.Paths
import kotlin.test.Test
import kotlin.test.assertEquals

class Q3Test {
    @Test
    fun `part 1 works`() {
        val filePath = javaClass.getResource("/q3.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(8085, part1(filePath!!))
    }

    @Test
    fun `part 2 works`() {
        val filePath = javaClass.getResource("/q3.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(2515, part2(filePath!!))
    }
}